RatScanner/View/FloatingTooltip.xaml.cs

```diff
@@ -0,0 +1,208 @@
+
+			var sortedHideoutItemRemaining = context.HideoutItemRemaining;
+			foreach (var hideoutItemRemaining in sortedHideoutItemRemaining)
+			{
+				var label = new Label();
+				label.Content = "[" + hideoutItemRemaining.Level + "] " + hideoutItemRemaining.Name + " " + hideoutItemRemaining.ItemCount + "x";
+				HideoutItems.Children.Add(label);
+			}
+		}
+
+
+		BarterItemList.Children.Clear();
+		if (context.LastItem.BartersUsing.Count > 0 && RcFloatingTooltip.ShowBarterInfo)
+		{
+			BarterItemList.Children.Add(new Separator());
+
+			Label barterTitle = new Label();
+			barterTitle.Foreground = Brushes.Orange;
+			barterTitle.Content = "Barters";
+			BarterItemList.Children.Add(barterTitle);
+
+			var item = context.LastItem;
+			var sortedBarterItems = item.BartersUsing.OrderBy(x => x.Trader.Name).ThenBy(x => x.Level);
+			foreach (var barter in sortedBarterItems)
+			{
+				StringBuilder sb = new StringBuilder();
+				sb.Append("[" + barter.Level + "] ");
+				sb.Append(barter.Trader.Name);
+				sb.Append(" => ");
+
+				for (var i = 0; i < barter.RequiredItems.Count; i++)
+				{
+					var requiredItem = barter.RequiredItems.ElementAt(i);
+					sb.Append(requiredItem.Item.Name).Append(" ").Append(requiredItem.Count).Append("x");
+
+					if (i < barter.RequiredItems.Count - 1)
+					{
+						sb.Append(" + ");
+					}
+				}
+
+				var rewardItem = barter.RewardItems.First();
+				sb.Append(" == ").Append(rewardItem.Item.Name).Append(" ").Append(rewardItem.Count).Append("x");
+
+				var label = new Label();
+				label.Content = sb.ToString();
+				BarterItemList.Children.Add(label);
+			}
+		}
+
+		CraftItemList.Children.Clear();
+		if (context.LastItem.CraftsUsing.Count > 0 && RcFloatingTooltip.ShowCraftsInfo)
+		{
+			CraftItemList.Children.Add(new Separator());
+
+			Label craftTitle = new Label();
+			craftTitle.Foreground = Brushes.Orange;
+			craftTitle.Content = "Crafts";
+			CraftItemList.Children.Add(craftTitle);
+
+			var item = context.LastItem;
+			var sortedCraftItem = item.CraftsUsing.OrderBy(x => x.Station.Name).ThenBy(x => x.Level);
+			foreach (var craft in sortedCraftItem)
+			{
+				StringBuilder sb = new StringBuilder();
+				sb.Append("[" + craft.Level + "] ");
+				sb.Append(craft.Station.Name);
+				sb.Append(" => ");
+
+				for (var i = 0; i < craft.RequiredItems.Count; i++)
+				{
+					var requiredItem = craft.RequiredItems.ElementAt(i);
+					sb.Append(requiredItem.Item.Name).Append(" ").Append(requiredItem.Count).Append("x");
+
+					if (i < craft.RequiredItems.Count - 1)
+					{
+						sb.Append(" + ");
+					}
+				}
+
+				var rewardItem = craft.RewardItems.First();
+				sb.Append(" == ").Append(rewardItem.Item.Name).Append(" ").Append(rewardItem.Count).Append("x");
+
+				var label = new Label();
+				label.Content = sb.ToString();
+				CraftItemList.Children.Add(label);
+			}
+		}
+
+		// TODO Joosep Sisas - 22.01.2024 - Maybe there is a better solution?
+		var screenBounds = Screen.PrimaryScreen.Bounds;
```

Comment:
It complains that the PrimaryScreen might be null. Is there a better Utility class to obtain screen size?

Reply:
There is `RatConfig.ScreenWidth` and `RatConfig.ScreenHeight` which use the config file from tarkov first and fallback to `Screen.PrimaryScreen.Bounds`. I believe it could become an issue when people play tarkov on their non-primary screen.

Reply:
Fixed